framework/decode/screenshot_handler.cpp

```diff
@@ -92,6 +95,10 @@ void ScreenshotHandler::WriteImage(const std::string&                      filen
 
     VkResult result = VK_SUCCESS;
 
+    // If both copy_scale and copy_width are provided, use copy_scale.
```

Comment:
I think the function that calls this function is a better location to do this conditional.  That is, receive only copy_width and copy_height in this function and calculate it in functions that call this function.